kale/pipeline/base_trainer.py

```diff
@@ -0,0 +1,168 @@
+# =============================================================================
+# Author: Xianyuan Liu, xianyuan.liu@outlook.com
+#         Haolin Wang, LWang0101@outlook.com
+# =============================================================================
+
+"""Classification of data
+The main structure is borrowed from kale.pipeline.domain_adapter.
+"""
+
+
+import pytorch_lightning as pl
+import torch
+
+from kale.predict import losses
+
+
+class BaseTrainer(pl.LightningModule):
+    """
+    Base class for classification models, based on pytorch lightning wrapper.
+    If you inherit from this class, a forward pass function must be implemented.
+    The structure is borrowed from kale.pipeline.domain_adapter.BaseAdaptTrainer.
+
+    Args:
+        optimizer (dict, None): optimizer parameters.
+        max_epochs (int): maximum number of epochs.
+        init_lr (float): initial learning rate. (default: 0.001)
+        adapt_lr (bool): adapt learning rate during training. (default: :obj:`False`)
```

Comment:
adapt is a version, should it be adaptive? https://discuss.pytorch.org/t/adaptive-learning-rate/320

Reply:
Yes. But to make it consistent with [domain_adapter](https://github.com/pykale/pykale/blob/main/kale/pipeline/domain_adapter.py#L232), it is better to change to `adapt_lr (bool): whether to use the schedule for the learning rate. Defaults to False.` and change all `(default: xxx)` to `Defaults to xxx`.

Btw, capitalization is not uniform in [BaseAdaptTrainer](https://github.com/pykale/pykale/blob/main/kale/pipeline/domain_adapter.py#L200). Shall we unify them in this PR?

Reply:
I suggest a separate PR since this one is large enough. Please put in to do or start the PR right away.